File: simnibs/examples/group_average_mni.m
%
%  This example wil go throgh simulations and calculate
%  the average a of the electric field norm in MNI space
%
% It is a follow-up to the "run_simulations" example

subjects = {'sub01', 'sub09', 'sub10', 'sub12', 'sub15'};
results_folder = fullfile('bipolar', 'mni_volumes');
field_name = 'normE';

mni_image_suffix = ['_TDCS_1_scalar_MNI_' field_name '.nii.gz'];

template_image = nifti_load(fullfile(subjects{1}, results_folder, [subjects{1} mni_image_suffix]));
field_avg = zeros(size(template_image.vol));

for i = 1:length(subjects)
    sub = subjects{i};
    % load the nifti image
    img = nifti_load(fullfile(pwd, sub, results_folder, [sub mni_image_suffix]));
    % accumulate values
    field_avg = field_avg + img.vol;
end

% calculate mean
field_avg = field_avg/length(subjects);

% save
avg_image = template_image;
avg_image.vol = field_avg;
nifti_save(avg_image, ['mean_' field_name '.nii.gz']);

File: simnibs/examples/analysis/group_average_mni.py
'''
    This example wil go throgh simulations and calculate
    the average of the electric field magnitude in MNI space

    It is a follow-up to the "run_simulations" example
'''

import os
import numpy as np
import nibabel as nib

subjects = ['sub01', 'sub09', 'sub10', 'sub12', 'sub15']
results_folder = os.path.join('bipolar', 'mni_volumes')
field_name = 'magnE'

mni_image_suffix = f'_TDCS_1_scalar_MNI_{field_name}.nii.gz'


# Go though each subject and load the images
images = []
for sub in subjects:
    images.append(
        nib.load(os.path.join(sub, results_folder, sub + mni_image_suffix))
    )

# calculate mean
avg = np.mean([img.get_fdata() for img in images], axis=0)

# save results
fn_out = f'mean_{field_name}.nii.gz'
nib.save(nib.Nifti1Image(avg, images[0].affine), fn_out)
print('Average field file: ', fn_out)
